.github/workflows/test.yml

```diff
@@ -73,73 +86,110 @@ jobs:
+          echo "matrix={\"include\":$(echo $matrix)}" >> $GITHUB_OUTPUT
+  
+  template-db:
     needs: check_changes
-    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'build')
+    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'template')
     uses: ./.github/workflows/template_test.yml
     secrets: inherit
-  data-library:
-    needs: check_changes
-    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'data-library')
-    uses: ./.github/workflows/data_library_test.yml
-    secrets: inherit
-  cbbr:
-    needs: check_changes
-    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'cbbr')
-    uses: ./.github/workflows/cbbr_test.yml
-    secrets: inherit
-  cpdb:
+
+  pytest:
+    name: Pytest - ${{ matrix.name }}
     needs: check_changes
-    if: github.event_name != 'pull_request' || contains(needs.check_changes.outputs.path_filters, 'cpdb')
-    uses: ./.github/workflows/cpdb_build.yml
-    with:
-      run_export: true
-    secrets: inherit
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.check_changes.outputs.matrix) }}
+    container: nycplanning/dev:latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ matrix.working-directory || '.' }}
+    env:
+      GHP_TOKEN: ${{ github.GITHUB_TOKEN }}
+      RECIPE_ENGINE: ${{ secrets.RECIPE_ENGINE }}
+      AWS_S3_BUCKET: edm-recipes
+      BUILD_ENGINE: ${{ secrets.EDM_DATA_QAQC_ENGINE }}
+      BUILD_ENGINE_SERVER: ${{ secrets.SQL_ENGINE_EDM_DATA_SERVER }}
+      BUILD_ENGINE_DB: edm-qaqc
+      ZAP_DOMAIN: ${{ secrets.ZAP_CRM_DOMAIN }}
+      TENANT_ID: ${{ secrets.ZAP_CRM_TENANT_ID }}
+      CLIENT_ID: ${{ secrets.ZAP_CRM_CLIENT_ID }}
+      SECRET: ${{ secrets.ZAP_CRM_SECRET }}
+      ZAP_ENGINE: ${{ secrets.SQL_ENGINE_EDM_DATA_SERVER }}/edm-zap
+      TEST_SCHEMA_SUFFIX: pr_${{ github.event.pull_request.number || 'workflow_dispatch' }}
+    steps:    
+    - uses: actions/checkout@v3
+    - name: Load Secrets
+      uses: 1password/load-secrets-action@v1
+      with:
+        export-env: true
+      env:
+        OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN }}
+        AWS_S3_ENDPOINT: "op://Data Engineering/DO_keys/AWS_S3_ENDPOINT"
+        AWS_SECRET_ACCESS_KEY: "op://Data Engineering/DO_keys/AWS_SECRET_ACCESS_KEY"
+        AWS_ACCESS_KEY_ID: "op://Data Engineering/DO_keys/AWS_ACCESS_KEY_ID"
+    - name: Run Container Setup
+      working-directory: ./
+      run: ./bash/docker_container_setup.sh
+
+    - name: Pytest
+      run: |
+        echo '${{ toJson(matrix.calls) }}' \
+          | sed '$ d;1d;s/^[[:space:]]*\"//;s/\",\{0,1\}[[:space:]]*$//'\
+          | while read call; do
+          python3 -m pytest $call
+        done
```

Comment:
this is a tad clunky. The matrix (json data in pytest.json) has this calls field which include the calls to be fed to pytest. The sed call is cleaning the string representation coming from toJson - removing first and last lines, cleaning whitespace and commas, then this gets iterated through. This was done because zap has two pytest calls.

It's sort of an unhappy middle ground though - the two pytest calls for zap are nicer to have as distinct steps for clarity when one fails, and if we're forced to do one step by the limitations of a matrix strategy, maybe we might as well just do a single pytest call (and just make this code a lot simpler)